ts/src/pro/krakenfutures.ts

```diff
@@ -0,0 +1,1283 @@
+                'requestLimit': 100, // per second
+                'watchTicker': {
+                    'method': 'ticker', // or ticker_lite
+                },
+                'watchTickers': {
+                    'method': 'ticker', // or ticker_lite
+                },
+                'fetchBalance': {
+                    'type': undefined,
+                },
+            },
+            'streaming': {
+                'keepAlive': 60000,
+            },
+        });
+    }
+
+    async authenticate (params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description authenticates the user to access private web socket channels
+         * @see https://docs.futures.kraken.com/#websocket-api-public-feeds-challenge
+         * @returns {object} response from exchange
+         */
+        this.checkRequiredCredentials ();
+        // Hash the challenge with the SHA-256 algorithm
+        // Base64-decode your api_secret
+        // Use the result of step 2 to hash the result of step 1 with the HMAC-SHA-512 algorithm
+        // Base64-encode the result of step 3
+        const url = this.urls['api']['ws'];
+        const messageHash = 'challenge';
+        const client = this.client (url);
+        let future = this.safeValue (client.subscriptions, messageHash);
+        if (future === undefined) {
+            const request = {
+                'event': 'challenge',
+                'api_key': this.apiKey,
+            };
+            const message = this.extend (request, params);
+            future = await this.watch (url, messageHash, message);
+            client.subscriptions[messageHash] = future;
+        }
+        return future;
+    }
+
+    async subscribePublic (name: string, symbols: string[], params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description Connects to a websocket channel
+         * @param {String} name name of the channel
+         * @param {[String]} symbols CCXT market symbols
+         * @param {Object} params extra parameters specific to the krakenfutures api
+         * @returns {Object} data from the websocket stream
+         */
+        await this.loadMarkets ();
+        const url = this.urls['api']['ws'];
+        const subscribe = {
+            'event': 'subscribe',
+            'feed': name,
+        };
+        const marketIds = [ ];
+        let messageHash = name;
+        if (symbols.length === 1) {
+            const symbol = symbols[0];
+            const marketId = this.marketId (symbol);
+            marketIds.push (marketId);
+            messageHash = messageHash + ':' + symbol;
+        }
+        for (let i = 0; i < symbols.length; i++) {
+            const symbol = symbols[i];
+            marketIds.push (this.marketId (symbol));
+        }
+        subscribe['product_ids'] = marketIds;
+        const request = this.extend (subscribe, params);
+        return await this.watch (url, messageHash, request, name);
```

Comment:
also, subscribing to different symbols simultaneously is not working `    await asyncio.gather(*[watch_ticker('ETH/USD:ETH'), watch_ticker('BTC/USD:BTC')])`

Reply:
Using the same `subscriptionHash` is the culprit here because it prevents the next subscription message to be sent, we should only do this when different topics share the same stream.

Reply:
> 

The 

> @samgermain what is missing here?

There's still those issues with `watchBalance` in the top comment

Reply:
> > @samgermain what is missing here?
> 
> I don't believe that anything is missing, as long as this comment is resolved [#17503 (comment)](https://github.com/ccxt/ccxt/pull/17503#discussion_r1178943433)

@samgermain I will check it again then